support-modules/acquisition-events/src/main/scala/com/gu/support/acquisitions/ga/GoogleAnalyticsServiceImpl.scala

```diff
@@ -0,0 +1,255 @@
+  def submit(acquisition: AcquisitionDataRow, gaData: GAData, maxRetries: Int)(implicit ec: ExecutionContext): EitherT[Future, List[GAError], Unit]
+}
+
+object MockGoogleAnalyticsService extends GoogleAnalyticsService {
+  def submit(acquisition: AcquisitionDataRow, gaData: GAData, maxRetries: Int)(implicit ec: ExecutionContext): EitherT[Future, List[GAError], Unit] =
+    EitherT.fromEither(Right(()))
+}
+
+class GoogleAnalyticsServiceImpl(client: OkHttpClient) extends GoogleAnalyticsService with LazyLogging {
+  private[ga] val gaPropertyId: String = "UA-51507017-5"
+  private[ga] val endpoint: HttpUrl = HttpUrl.parse("https://www.google-analytics.com")
+
+  private[ga] def buildBody(acquisition: AcquisitionDataRow, gaData: GAData)(implicit ec: ExecutionContext): EitherT[Future, BuildError, RequestBody] = EitherT {
+    getAnnualisedValue(acquisition)
+      .leftMap(error => {
+        logger.warn(s"Couldn't retrieve annualised value for this acquisition: $error")
+        0D // We still want to record the acquisition even if we can't get the AV
+      })
+      .merge[Double]
+      .map(av => buildPayload(acquisition, av, gaData))
+      .map { maybePayload =>
+        maybePayload.map { payload =>
+          logger.debug(s"GA payload: $payload")
+          RequestBody.create(null, payload)
+        }
+      }
+  }
+
+  private[ga] def getAnnualisedValue(acquisition: AcquisitionDataRow)(implicit ec: ExecutionContext): EitherT[Future, String, Double] = {
+    acquisition.amount match {
+      case Some(amount) =>
+        val acquisitionModel = AcquisitionModel(amount.toDouble,
+          acquisition.product.value,
+          acquisition.currency.iso,
+          acquisition.paymentFrequency.value,
+          acquisition.paymentProvider.map(_.toString),
+          acquisition.printOptions.map(printOptions => PrintOptionsModel(printOptions.product.value, printOptions.deliveryCountry.alpha2))
+        )
+        EitherT(AnnualisedValueService.getAsyncAV(acquisitionModel, "ophan"))
+
+      case None => EitherT.fromEither[Future](Left("No amount for acquisition"))
+    }
+  }
+
+  private[ga] def getSuccessfulSubscriptionSignUpMetric(conversionCategory: ConversionCategory) =
+    conversionCategory match {
+      case ConversionCategory.ContributionConversion => ""
+      case _ => "1"
+    }
+
+  private[ga] def buildPayload(acquisition: AcquisitionDataRow, annualisedValue: Double, gaData: GAData, transactionId: Option[String] = None): Either[BuildError, String] = {
+    val tid = transactionId.getOrElse(UUID.randomUUID().toString)
+    val goExp = buildOptimizeTestsPayload(acquisition.abTests)
+
+    // clientId cannot be empty or the call will fail
+    sanitiseClientId(gaData.clientId).map {
+      clientId =>
+        val productName = getProductName(acquisition)
+        val conversionCategory = getConversionCategory(acquisition)
+        val productCheckout = getProductCheckout(acquisition)
+        val body = Map(
```

Comment:
```suggestion
        val body = Map[String, String](
```
adding that in intellij shows the first line which has the issues so it's easier to resolve, there are actually a few